src/ORM/DataObject.php

```diff
@@ -3175,22 +3175,24 @@ public static function reset()
     /**
      * Return the given element, searching by ID
      *
-     * @param string $callerClass The class of the object to be returned (optional if called from extending class)
-     * @param int $id The id of the element
-     * @param boolean $cache See {@link get_one()}
+     * @param string|int $callerClassOrID The class of the object to be returned (optional if called from extending class)
+     * @param int|bool $id The id of the element or bool to represent cache if ID is passed as fist argument
+     * @param bool $cache See {@link get_one()}
      *
-     * @return DataObject The element
+     * @return static The element
      */
-    public static function get_by_id($callerClass, $id = null, $cache = true)
+    public static function get_by_id($callerClassOrID, $id = null, $cache = true)
     {
-        if(is_numeric($callerClass)) {
+        if (is_numeric($callerClassOrID)) {
             // Alternate signature to allow calling as MyDataObject::get_by_id($id)
-            $cache = is_null($id) ? true : !!$id;
-            $id = $callerClass;
+            $id = $callerClassOrID;
+            $cache = is_null($id) ? true : (bool)$id;
```

Comment:
@dhensby Isn't this in the wrong order now? You're first assigning a new value to $id, and then using $id to set $cache.

Reply:
good point - fixed

Reply:
Don't use argument re-assigment on $id. That's one of my original concerns (and demonstrates why this is bad practice). :P